### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by donbowen** [LeDataSciFi/the-fin388-project-team-x#2]
Hi - you can submit revisions until 11pm Wednesday.

A smattering of thoughts as I read
- The smaller questions/data we need are formatted funny, and unlike other lists on the document. I'm guessing you indented those two lists.
- The "smaller" questions are good, but the "bigger problem" is worded oddly. Given your aims, you don't really have an "Initial hypothesis" to test - you just want to develop an accurate prediction. You could list as your hypothesis which models would do best.
- I like the restriction to tech stocks. It might help in the analysis and write up. Please try to exploit the restriction. E.g., you can look for factors that might be ignored by people trying to predict the returns of all stocks, or just predict utility companies.
- Why 2012-2021. Seems incredibly random. Go longer! (Or explain why 2012.)
- What in the heck? Where did "twitter mentions" come from? Another class?

## The main thing to revise

Ok, so the main thing I'd like to get more clarity on is the description of the models in the "Neccesary Data" section. 

The three models are doing different things and its not clear from the write-up why you choose each, and how it might be used to make predictions about Tesla.** And the descriptions are VERY vague. Like... I have almost no clue what you intend to do with each. I can guess, maybe.

1. The regression model covers NASDAQ firms (not the top 50 tech firms). Why are you using 10 year rates and twitter for this? And what is the variable "stock prices"? NASDAQ after all, *is stock prices*. How will you apply it to Tesla? As it is written, this model will make predictions for Tesla as though it is the NASDAQ itself. 
2. Time series model. Of what "stock prices"? What time series model? How will it apply to Tesla?
3. Deep learning model. Which model? What is the y variable? Do you really plan to predict returns (presumably that's the y variable) with high and low? Returns of what assets? How are these variables useful or possible? (If I know the high and the low of the day before it starts, I would be able to make a profit trading, of course. But this is impossible to know when you make your trades.)

**Comment by chw221** [comment]
Based on the feedback provided, we have made significant revisions to our proposal and changed some focus of our topic. The updated details could be found in our newly uploaded proposal in readme.

 1. Now for the regression model, we have dropped Nasdaq, 10-year rate, and Twitter variables. The new variables are lithium battery prices (LIT ETF price), nanochip price (SMH ETF price), and Brent index for cruise oil prices. These variables may not have the most optimal predictive power and may generate relatively low prediction accuracy, but we are curious to see how these industry supply-related variables could affect the EV industry and EV stock returns. For example, we assume there could be a strong correlation between battery/semiconductor price and EV component supply, or there could be a correlation between oil price and EV segment as oil is required for manufacturing, distributing, and electricity-generating in the EV supply chain, although EV itself does not require oil (gasoline).

 2. The time series model that we are going to implement in our project is called Prophet. And we plan to use Prophet to predict stock returns of four EV companies, including TSLA, KNDI, AYRO, and WKHS, and one traditional car manufacturer, F. Time series model is considered useful because the model is capable to predict future values based on previously observed values on an ordinal seasonally and timely basis. The previously observed values are the high, low, close, volume, and adjusted close of our targeted companies’ stock price data, and the future value that Prophet is going to predict is the future stock returns of those five companies.
 
3. The deep learning model is the LSTM, Long Short Term Memory model. LSTM is a machine learning model that involves recurrent neural networks. We believe LSTM is potentially effective in predicting stock returns as LSTM has the capability to analyze existing observation patterns and make sequential predictions of future values. The Y variable is the stock return, and the X variables that we are going to input for the LSTM model are similar to Prophet’s, which are high, low, close, volume, and adjusted close of the targeted companies stock prices. After completing thorough research with online resources, we currently do not find any variables that can better train the LSTM other than using stock price data. But we may try to include more useful or effective variable inputs throughout the actual modeling of the LSTM.
 
Thank you very much for your constructive suggestion and feedback. We will continue to improve our project details.

**Comment by donbowen** [comment]
> Based on the feedback provided, we have made significant revisions to our proposal and changed some focus of our topic. The updated details could be found in our newly uploaded proposal in readme.

Very good. This was one of the bigger revisions, but it's markedly better. Go forth and conquer! I do have some final feedback below. 

>  1. Now for the regression model, we have dropped Nasdaq, 10-year rate, and Twitter variables. The new variables are lithium battery prices (LIT ETF price), nanochip price (SMH ETF price), and Brent index for cruise oil prices. 

Please also include the "Fama/French 5 Factors (2x3) [Daily]", as described [here](https://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html). You can download via pandas datareader as in examples. 

>  2. The time series model that we are going to implement in our project is called Prophet... **The previously observed values are the high, low, close, volume, and adjusted close** of our targeted companies’ stock price data, and the future value that Prophet is going to predict is the future stock returns of those five companies.

**I'm confused on this aspect.** I thought the Prophet and LSTM models would price future stock returns using only the past stock returns? They'll use past open/close/volume??? (Please confirm yes/no)

Also, you alluded to this in the write up, but we talked about it and it is utterly essential to reiterate: You predict *tomorrow's* returns with _**variables from today, and the past**_ 

> Thank you very much for your constructive suggestion and feedback. We will continue to improve our project details.

Again, very good improvement! (Grade: A)

**Comment by sellalin** [comment]
No, the LSTM and Prophet actually don't use past open/close/volume. They predict only based on the past stock returns. We will make some changes to clarify this point.

**Comment by donbowen** [comment]
Good, thanks!
